packages/datadog-instrumentations/src/mocha.js

```diff
@@ -75,34 +95,46 @@ function mochaHook (Runner) {
     }))
 
     this.on('suite', function (suite) {
-      if (suite.root) {
+      if (suite.root || !suite.tests.length) {
         return
       }
-      const asyncResource = new AsyncResource('bound-anonymous-fn')
-
-      testSuiteToAr.set(suite, asyncResource)
-
-      asyncResource.runInAsyncScope(() => {
-        testSuiteStartCh.publish(suite)
-      })
+      let asyncResource = testFileToSuiteAr.get(suite.file)
+      if (!asyncResource) {
+        asyncResource = new AsyncResource('bound-anonymous-fn')
+        testFileToSuiteAr.set(suite.file, asyncResource)
+        asyncResource.runInAsyncScope(() => {
+          testSuiteStartCh.publish(suite)
+        })
+      }
     })
 
     this.on('suite end', function (suite) {
       if (suite.root) {
         return
       }
+      const suitesInTestFile = suitesByTestFile[suite.file]
+
+      const isLastSuite = suitesInTestFile.filter(suite => suite._ddFinished).length === suitesInTestFile.length - 1
+      suite._ddFinished = true
```

Comment:
This should be a Set instead, or if possible maybe even a counter.

Reply:
yeah using a counter makes sense. I'll do the change